### ISSUE TITLE
Support initialisation with AppOptions

### ISSUE THREAD

**Issue by dri-richard** [defold/extension-firebase#4]
firebase::App::Create() has an overload that takes an object of type AppOptions to permit Firebase to be initialized with non-default values. 
We'd like to use it to permit selection of Firebase projects for different environments (dev, staging etc) from a boot screen.

I'm happy to take this on myself, but could do with some guidance on the Lua API, as AppOptions has 6 string members.

**Comment by britzl** [comment]
> I'm happy to take this on myself, but could do with some guidance on the Lua API, as AppOptions has 6 string members.

We should change firebase.init() to take an optional Lua table with options, ie:

```lua
local options = {
  api_key = "a",
  app_id = "b",
  project_id = "c",
}
firebase.init(options)
```

The extension code will look something like this:

```c++
	AppOptions options;
	if (!lua_isnil(L, 1)) {
		luaL_checktype(L, 1, LUA_TTABLE);
		lua_pushvalue(L, 1);
		lua_pushnil(L);
		while (lua_next(L, -2)) {
			const char* attr = lua_tostring(L, -2);
			if (strcmp(attr, "api_key") == 0)
			{
				options.set_api_key(luaL_checkstring(L, -1));
			}
			else if (strcmp(attr, "app_id") == 0)
			{
				options.set_app_id(luaL_checkstring(L, -1));
			}
			else if (strcmp(attr, "project_id") == 0)
			{
				options.set_project_id(luaL_checkstring(L, -1));
			}
			lua_pop(L, 1);
		}
		lua_pop(L, 1);
	}
```

**Comment by britzl** [comment]
Closed via #5
